### ISSUE TITLE
Copy model using Python binding without save/load

### ISSUE THREAD

**Issue by danielhers** [clab/dynet#145]
In order to do early stopping I am training my model on my training set and evaluating on the development set after each iteration, eventually keeping only the model with the highest development set score.
For that I need to keep a copy of my model, which is not updated as I keep training the original.
Currently I save the model to a file and immediately load it to achieve this functionality, but it would be more efficient to use a copy constructor or something like that.

**Comment by neubig** [comment]
I think having the ability to copy the model would be nice, so I'll mark this an enhancement.

But why don't you just save the model every time you achieve the best development score? I don't understand the necessity to keep it in memory.

**Comment by danielhers** [comment]
Thanks. I want to evaluate the best model on the test set at the end of the run. But I guess what you suggest does make more sense than keeping two models in memory.
